index.bs

```diff
@@ -90,6 +100,37 @@ The exact type of the `globalScope` can vary across runtimes. Most Web Platform
 
 With many runtimes, adding a new global-scoped property can introduce breaking changes when the new global conflicts with existing application code. Many Web Platform APIs define global properties using the `readonly` attribute. To avoid introducing breaking changes, runtimes conforming to this specification MAY choose to ignore the `readonly` attribute for properties being added to the global scope.
 
+Requirements for EcmaScript modules {#esm-requirements}
+=======================================================
+
+[=Web-interoperable runtimes=] which support EcmaScript modules must implement the following ECMA-262 host hooks as follows:
+
+: [$HostLoadImportedModule$]
+:: This host hook must be implemented [[HTML#hostloadimportedmodule|as defined in the HTML spec]], except that they may pass a [=perform the fetch hook=] algorithm to [=fetch a single imported module script=] even when <var ignore="">loadState</var> is undefined. (See also [[#custom-module-loading]].)
```

Comment:
Where can this `perform the fetch hook` be obtained from? Somewhere globally available, or can it be potentially different per module?

Reply:
The intention was to have a single "perform the fetch hook" algorithm per runtime that needs or supports module loading customizations. But the intention was also for this spec change to be generic enough that it could cover npm imports and Node.js's custom loaders, but now that I look into it, even a per-module hook isn't enough for that because it doesn't override module resolution. I will think of other ways to make these customizations possible.